diff --git a/src/panfrost/vulkan/csf/panvk_vX_cmd_dispatch.c b/src/panfrost/vulkan/csf/panvk_vX_cmd_dispatch.c
index 3ce44dd..eb454f3 100644
--- a/src/panfrost/vulkan/csf/panvk_vX_cmd_dispatch.c
+++ b/src/panfrost/vulkan/csf/panvk_vX_cmd_dispatch.c
@@ -50,7 +50,7 @@ void panvk_per_arch(cmd_signal_barrier)(
                PANVK_SUBQUEUE_COMPUTE * sizeof(struct panvk_cs_sync64));
       cs_move64_to(b, add_val, 1);
       panvk_instr_sync64_add(cmdbuf, PANVK_SUBQUEUE_COMPUTE, true,
-                             MALI_CS_SYNC_SCOPE_CSG, add_val, sync_addr,
+                             cmdbuf->sync_scope, add_val, sync_addr,
                              cs_defer_indirect());
 #else
       struct cs_index sync_addr = cs_scratch_reg64(b, 0);
@@ -68,7 +68,7 @@ void panvk_per_arch(cmd_signal_barrier)(
 
       cs_match_iter_sb(b, x, iter_sb, cmp_scratch) {
          panvk_instr_sync64_add(cmdbuf, PANVK_SUBQUEUE_COMPUTE, true,
-                                MALI_CS_SYNC_SCOPE_CSG, add_val, sync_addr,
+                                cmdbuf->sync_scope, add_val, sync_addr,
                                 cs_defer(SB_WAIT_ITER(x), SB_ID(DEFERRED_SYNC)));
       }
 #endif
diff --git a/src/panfrost/vulkan/csf/panvk_vX_cmd_event.c b/src/panfrost/vulkan/csf/panvk_vX_cmd_event.c
index 728a2eb..d4f48df 100644
--- a/src/panfrost/vulkan/csf/panvk_vX_cmd_event.c
+++ b/src/panfrost/vulkan/csf/panvk_vX_cmd_event.c
@@ -50,7 +50,7 @@ panvk_per_arch(CmdResetEvent2)(VkCommandBuffer commandBuffer, VkEvent _event,
 
          cs_default(b) {
             cs_move32_to(b, seqno, 0);
-            cs_sync32_set(b, false, MALI_CS_SYNC_SCOPE_CSG, seqno, sync_addr,
+            cs_sync32_set(b, false, cmdbuf->sync_scope, seqno, sync_addr,
                           cs_defer(sb_mask | SB_MASK(DEFERRED_FLUSH),
                                    SB_ID(DEFERRED_SYNC)));
          }
@@ -97,7 +97,7 @@ panvk_per_arch(CmdSetEvent2)(VkCommandBuffer commandBuffer, VkEvent _event,
             }
 
             cs_move32_to(b, seqno, 1);
-            cs_sync32_set(b, false, MALI_CS_SYNC_SCOPE_CSG, seqno, sync_addr,
+            cs_sync32_set(b, false, cmdbuf->sync_scope, seqno, sync_addr,
                           cs_defer(sb_mask | SB_MASK(DEFERRED_FLUSH),
                                    SB_ID(DEFERRED_SYNC)));
          }
diff --git a/src/panfrost/vulkan/csf/panvk_vX_cmd_query.c b/src/panfrost/vulkan/csf/panvk_vX_cmd_query.c
index bcd8388..98aaa6f 100644
--- a/src/panfrost/vulkan/csf/panvk_vX_cmd_query.c
+++ b/src/panfrost/vulkan/csf/panvk_vX_cmd_query.c
@@ -333,7 +333,7 @@ panvk_cmd_end_occlusion_query(struct panvk_cmd_buffer *cmd,
    /* Signal the query syncobj after the flush is effective. */
    cs_move32_to(b, val, 1);
    cs_move64_to(b, oq_syncobj, panvk_query_available_dev_addr(pool, query));
-   cs_sync32_set(b, true, MALI_CS_SYNC_SCOPE_CSG, val, oq_syncobj,
+   cs_sync32_set(b, true, cmd->sync_scope, val, oq_syncobj,
                  cs_defer(SB_MASK(DEFERRED_FLUSH), SB_ID(DEFERRED_SYNC)));
 }
 
@@ -713,7 +713,7 @@ panvk_cmd_end_prims_generated_query(
    /* Signal the query syncobj after the flush is effective. */
    cs_move32_to(b, val, 1);
    cs_move64_to(b, query_syncobj, panvk_query_available_dev_addr(pool, query));
-   cs_sync32_set(b, true, MALI_CS_SYNC_SCOPE_CSG, val, query_syncobj,
+   cs_sync32_set(b, true, cmd->sync_scope, val, query_syncobj,
                  cs_defer(SB_MASK(DEFERRED_FLUSH), SB_ID(DEFERRED_SYNC)));
 }
 
